internal/code-usage-scan-tool/src/main/java/org/wso2/security/tools/util/DatabaseUtils.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.security.tools.util;
+
+import com.mongodb.BasicDBObject;
+import com.mongodb.BasicDBObjectBuilder;
+import com.mongodb.DB;
+import com.mongodb.DBCollection;
+import com.mongodb.DBCursor;
+import com.mongodb.DBObject;
+import com.mongodb.MongoClient;
+import com.mongodb.MongoCredential;
+import com.mongodb.ServerAddress;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.security.tools.model.MethodReference;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/**
+ * DatabaseUtils - Handles all the database  related functionality.
+ */
+public class DatabaseUtils {
+
+    private static final Logger log = LoggerFactory.getLogger(DatabaseUtils.class);
+    private static final String USERNAME = "mgdb";
```

Comment:
Credentials should not be hard coded. In this case, we'll have to build the code when the credentials change. This should be read from a property file.